### PR TITLE
featureimage/IPMPROG-2969-metric-storage

### PR DESCRIPTION
**Pull request by FrancoisRegisDegott-eaton**
NOTICE: based on [featureimage/IPMVAL-3756-endu-graphs](https://github.com/42ity/fty-metric-store/tree/featureimage/IPMVAL-3756-endu-graphs)

### CODE REVIEW

File: src/converter2.cc [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        }
+        catch (const std::out_of_range& e) {
+            error = "std::out_of_range, e: " + std::string{e.what()};
+        }
+        catch (...) {
+            error = "unexpected exception";
+        }
+
+        if (!error.empty()) {
+            logError("parse double value has failed ('" + string + "'), {}", error);
```
Comment (perrettecl) [inline src/converter2.cc]:
```suggestion
            logError("parse double value has failed ('{}'), {}", string, error);
```

**Review by perrettecl — CHANGES_REQUESTED** [review]
Renaming "string" by "str" or something even more meaning full could be good, as it's kind of confusing.

File: src/converter2.cc (hunk 1 above)
Comment (FrancoisRegisDegott-eaton) [reply src/converter2.cc]: done

File: src/persistance.cc [hunk 2]
```diff
@@ -208,11 +208,15 @@ m_msrmnt_tpc_id_t prepare_topic(tntdb::Connection& conn, const char* topic, cons
...
+        //if (conn.lastInsertId() != m_msrmnt_tpc_id_t(conn.lastInsertId())) {
+        //    logError("conn.lastInsertId(): {} neq. {}", conn.lastInsertId(), m_msrmnt_tpc_id_t(conn.lastInsertId()));
+        //}
+
         m_msrmnt_tpc_id_t topic_id = m_msrmnt_tpc_id_t(conn.lastInsertId());
         if (topic_id != 0) {
             //log_trace("inserted topic %s, topic_id %u", topic, topic_id);
         } else {
-            log_error("topic %s not inserted", topic);
+            log_error("topic '%s' not inserted", topic);
```
Comment (stark-dev) [inline src/persistance.cc:224]: Not sure if doable here, but the new interface for logging supports `std::string` and may be easier to maintain, as we have currently lot of blob data in the log caused by C++ strings not converted to plain char arrays

**Review by stark-dev — DISMISSED** [review]

File: src/persistance.cc (hunk 2 above)
Comment (perrettecl) [inline src/persistance.cc:224]: Maybe like this ?

```suggestion
            logError("topic '{}' not inserted", topic);
```

**Review by perrettecl — APPROVED** [review]

File: src/persistance.cc (hunk 2 above)
Comment (FrancoisRegisDegott-eaton) [reply src/persistance.cc:224]: i agree, and shall be done globally... but it is not really the goal of the PR ;)

**Pull request merged**